src/s3/commands/downloadFileAs.ts

```diff
@@ -119,9 +119,13 @@ export async function downloadFileAsCommand(
     const sourcePath = readablePath(node)
 
     await telemetry.s3_downloadObject.run(async () => {
-        const saveLocation = await promptForSaveLocation(file.name, window)
+        const downloadPath = getDefaultDownloadPath()
+
+        const saveLocation = await promptForSaveLocation(file.name, window, downloadPath)
         if (!saveLocation) {
             throw new CancellationError('user')
+        } else {
+            setDefaultDownloadPath(saveLocation.path)
```

Comment:
Don't need `else` when the other branch exits the scope (`throw`, `return`, etc.). Please use this pattern for all future code.

```suggestion
        setDefaultDownloadPath(saveLocation.path)
```